sdk/identity/Azure.Identity/src/ManagedIdentitySource.cs

```diff
@@ -87,16 +87,17 @@ protected virtual async ValueTask<AccessToken> HandleResponseAsync(
             if (response.IsError)
             {
                 string content = string.Empty;
+
                 try
                 {
                     content = response.Content.ToString();
                     using JsonDocument json = async
                     ? await JsonDocument.ParseAsync(response.ContentStream, default, cancellationToken).ConfigureAwait(false)
                     : JsonDocument.Parse(response.ContentStream);
                 }
-                catch (JsonException)
+                catch (Exception)
                 {
-                    // If the response is not json, it is not the IMDS and it should be treated as CredentialUnavailable
+                    // If the response is not json or the Content was null, it is not the IMDS and it should be treated as CredentialUnavailable
                     throw new CredentialUnavailableException(UnexpectedResponse, new Exception(content));
```

Comment:
Would it make sense to add a filter on this rather than catching all exceptions? or do we want all unhandled exceptions to result in a `CredentialUnavailableException` rather than `AuthenticationFailedException`?

Reply:
In this case, I think it makes sense to catch all exceptions that might occur during json deserialization. DAC makes less guarantees about resiliency and should err on the side of `CredentialUnavailableException`.